### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by virbas**

### CODE REVIEW

**Comment by virbas** [comment]
- update deprecated code
- make sure all examples work with v1
- generalize XR8Image and XR8VPS events

File: ar-tracking/src/tracking-mode.ts [hunk 1]
```diff
@@ -110,8 +147,51 @@ export interface VPSTrackingMode extends ITrackingMode {
     readonly onWaySpotLost: Emitter<[event: VPSWayPointEvent]>;
 }
 
-export interface ImageScanningEvent {}
-export interface ImageTrackedEvent {}
+export interface ImageScanningEvent {
+    imageTargets: {
+        name: string; // image name
```
Comment (Squareys) [inline ar-tracking/src/tracking-mode.ts:152]:
```suggestion
        /** Image name */
        name: string;
```

File: examples/absolute-scale/js/button-end-ar-session.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * Shows a simple HTML buttton when ARSession has started.
+ * When clicked - kill any running AR session.
+ */
+import {Component} from '@wonderlandengine/api';
+import {ARSession} from '@wonderlandengine/ar-tracking';
+
+export class ButtonEndARSession extends Component {
+    static TypeName = 'button-end-ar-session';
+
+    xrEndButton!: HTMLButtonElement;
```
Comment (Squareys) [inline examples/absolute-scale/js/button-end-ar-session.ts:12]: I wonder if you could maybe create a package `ar-tracking-example-utils` that is shared by all? Or put it in common-components? Kind of got used to that folder now :)

**Review by Squareys — COMMENTED** [review]
Very nice cleanup! Went through and found some smaller things you could add to the cleanup, but looks pretty great already.

File: examples/absolute-scale/js/button-end-ar-session.ts (hunk 2 above)
Comment (virbas) [reply examples/absolute-scale/js/button-end-ar-session.ts:12]: Yeah, that's a temporary workaround for at least make the examples run.
nightly-20230508 would crash on startup with commont-components. (files outside the root of the project)

File: examples/SLAM-camera/js/button-end-ar-session.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * Shows a simple HTML buttton when ARSession has started.
+ * When clicked - kill any running AR session.
+ */
+import {Component} from '@wonderlandengine/api';
+import {ARSession} from '@wonderlandengine/ar-tracking';
+
+export class ButtonEndARSession extends Component {
+    static TypeName = 'button-end-ar-session';
+
+    xrEndButton!: HTMLButtonElement;
```
[the remark this answers is not in the document]
Comment (virbas) [reply examples/SLAM-camera/js/button-end-ar-session.ts:12]: @Squareys 
So I'll just run the logic in init().
Or is init() still called by the editor?

**Pull request merged**